rawhttp-core/src/main/java/rawhttp/core/RawHttp.java

```diff
@@ -276,7 +276,14 @@ public FramedBody getFramedBody(StartLine startLine, RawHttpHeaders headers) {
      * @param headers HTTP request's headers
      * @return true if the headers indicate the request should have a body, false otherwise
      */
-    public static boolean requestHasBody(RawHttpHeaders headers) {
+    public static boolean requestHasBody(RawHttpHeaders headers, RequestLine requestLine) {
+        // As per RFC-7231 section 4.3.6:
+        // A payload within a CONNECT request message has no defined semantics;
+        // sending a payload body on a CONNECT request might cause some existing
+        // implementations to reject the request.
+        if ("CONNECT".equalsIgnoreCase(requestLine.getMethod())) {
+            return false;
```

Comment:
This is unnecessary. The RFC does not forbid a body, it only says there's no semantic value to it. It's the same phrasing as for GET requests... and it's fairly common to use bodies in GET requests. In any case, I don't think this is needed for supporting CONNECT.